addons/website/models/website_snippet_filter.py

```diff
@@ -86,19 +86,24 @@ def _prepare_values(self, limit=None, search_domain=None):
         if self.filter_id:
             filter_sudo = self.filter_id.sudo()
             domain = filter_sudo._get_eval_domain()
+            filter_context = literal_eval(filter_sudo.context)
             if 'website_id' in self.env[filter_sudo.model_id]:
                 domain = expression.AND([domain, self.env['website'].get_current_website().website_domain()])
             if 'is_published' in self.env[filter_sudo.model_id]:
                 domain = expression.AND([domain, [('is_published', '=', True)]])
             if search_domain:
                 domain = expression.AND([domain, search_domain])
             try:
-                records = self.env[filter_sudo.model_id].search(
+                records = self.env[filter_sudo.model_id].with_context(**filter_context).search(
                     domain,
                     order=','.join(literal_eval(filter_sudo.sort)) or None,
                     limit=limit
                 )
-                return self._filter_records_to_values(records)
+                values = self._filter_records_to_values(records)
+                if records.env.context.get('action_rerender'):
+                    for data in values:
+                        data['_action_rerender'] = True
```

Comment:
will mak emore sense to got it in _filter_Records_to_values method no??

And need to decide if it is `rerender on add_to_cart`, or `rerender on action`...

rerender on action lgtm, but need to be done for each action wishlist, delete, ... and not only on add2Cart

Here it seems to be an on_add2cart with a name on_action

Reply:
On the top of my head, I think the reason I've set it here is so that we don't call filter_id.sudo() twice, but I could pass it as an argument. to the _filter_records_to_value method. Or maybe make the context of website_snippet_filter to match the context defined in ir.filter, so the infos defined there can be used anywhere ? Still a bit new with the python framework so I don't know what's best 😅 

For the name, I think it might be nice to make sure it's available for any action, since we probably don't want the widget to refresh itself when we just add and item to a wishlist. Right we don't have a template that includes a wishlist button but maybe that'll change? Although maybe it's not needed for this fix. Let me know what you think 😄

Reply:
Just thought about it a bit more and it's probably not a good idea to have something generic for this. If we implement a wishlist button on a template we'll probably need some extra code anyway. I'm going to work on cleaning up this PR so that it mostly impacts website_sale.